src/proxy/inbound.rs

```diff
@@ -363,12 +363,26 @@ impl Inbound {
         upstream: &Workload,
         conn: &Connection,
     ) -> (bool, bool) {
-        let has_waypoint = !upstream.waypoint_addresses.is_empty();
-        for i in upstream.waypoint_addresses.iter() {
-            if workloads.fetch_workload(i).await.map(|w| w.identity()) == conn.src_identity {
-                return (has_waypoint, true);
+        let has_waypoint = upstream.waypoint.address.is_some();
+        let wp_ip = match upstream.waypoint.address.as_ref() {
+            Some(addr) => match addr {
+                gatewayaddress::Address::IP(wp_ip) => wp_ip,
+                gatewayaddress::Address::Hostname(_) => return (has_waypoint, false),
+            },
+            None => return (has_waypoint, false),
+        };
```

Comment:
**nit:** `has_waypoint` can probably just be hard coded to `true` instead of checking `is_some()`. The next line is matching on the same field and there is a `None` arm that handles the case where `has_waypoint` is false by returning immediately. That arm could just be `return (false, false)`

Reply:
this is going to change (no more `is_some` on address) once i implement https://github.com/istio/ztunnel/pull/486#discussion_r1157494478

Reply:
implemented in https://github.com/istio/ztunnel/pull/486/commits/6d7827f3110d69e6b1eb94d5975120c92fa3cf79 and https://github.com/istio/ztunnel/pull/486/commits/38d0a5394347413e198ce5afd6e82ce1f07c1913

Reply:
this function has been updated, closing this as dated